Mathlib/Tactic/Linter/Style.lean

```diff
@@ -467,4 +469,50 @@ initialize addLinter doubleUnderscore
 
 end Style.nameCheck
 
+/-! # The "openClassical" linter -/
+
+/-- The "openClassical" linter emits a warning on `open Classical` statements which are not
+scoped to a single declaration. A non-scoped `open Classical` can hide that some theorem statements
+would be better stated with explicit decidability statements.
+-/
+register_option linter.style.openClassical : Bool := {
+  defValue := false
+  descr := "enable the openClassical linter"
+}
+
+namespace Style.openClassical
+
+/-- If `stx` is syntax describing an `open` command, `extractOpenNames stx`
+returns an array of the syntax corresponding to the opened names,
+omitting any renamed or hidden items. -/
```

Comment:
It's not so clear from the name or docstring that this only works for `open`, not `open ... in ...`. How about:

```suggestion
/-- If `stx` is syntax describing an `open` command, `extractOpenNames stx`
returns an array of the syntax corresponding to the opened names,
omitting any renamed or hidden items.

This only checks independent `open` commands: for `open ... in ...` commands,
this linter returns an empty array.
-/
```

Reply:
Good point, thanks!